docs/manual/faq.rst

```diff
@@ -75,4 +75,52 @@ LibreOffice menus don't appear or don't stay visible
 A workaround for problem with the mouse in libreoffice is setting the environment variable »SAL_USE_VCLPLUGIN=gen«.
 It is dependet on your system configuration where to do this. e.g. ArchLinux with libreoffice-fresh in /etc/profile.d/libreoffice-fresh.sh.
 
+How can I get my groups to stick to screens?
+============================================
 
+This behaviour can be replicated by configuring your keybindings to not move
+groups between screens. For example if you want groups ``"1"``, ``"2"`` and
+``"3"`` on one screen and ``"q"``, ``"w"``, and ``"e"`` on the other, instead
+of binding keys to ``lazy.group[name].toscreen()``, use this:
+
+.. code-block:: python
+
+    def go_to_group(name: str) -> Callable:
+        def _inner(qtile: Qtile) -> None:
+            if len(qtile.screens) == 1:
+                qtile.groups_map[name].cmd_toscreen()
+                return
+
+            if name in '123':
+                qtile.focus_screen(0)
+                qtile.groups_map[name].cmd_toscreen()
+            else:
+                qtile.focus_screen(1)
+                qtile.groups_map[name].cmd_toscreen()
+
+        return _inner
+
+    for i in groups:
+        keys.append(Key([mod], i.name, lazy.function(go_to_group(i.name))))
+
+If you use the ``GroupBox`` widget you can make it reflect this behaviour:
+
+.. code-block:: python
+
+    groupbox1 = widget.GroupBox(visible_groups=['1', '2', '3'])
+    groupbox2 = widget.GroupBox(visible_groups=['q', 'w', 'e'])
+
+And if you jump between having single and double screens then modifying the
+visible groups on the fly may be useful:
+
+.. code-block:: python
+
+   @hook.subscribe.screen_change
+   async def _(_):
+       await asyncio.sleep(0.5)  # Why is this necessary?
```

Comment:
Oh very interesting - yes that would likely fix this! Unfortunately I won't have an external monitor for a week or two so can't test... I think you're absolutely right though.

Reply:
Shall we hold back merging this until then?

Reply:
Just checking all the code again that must work, because it just relies on `qtile.screens` updating before this hooked function, and that happens upon `screen_change`. It was just the ordering that was wrong for me. I'd be happy to update this and call it a day.

Reply:
Yeah, I was pretty certain it had to be the same issue. 

Make the change and I'll merge.

Reply:
Cool, done